http://developer.pidgin.im/viewmtn/revision/info/e05d46b1a88dfa81c3cc655bf9ab3e90bb6be828
Fix an assertion failure/unlikely crash when:
1. Open a purple_notify_formatted() dialog, such as the QQ about window
or the AIM "who am I waiting on authorization from?" dialog
2. Close the dialog with the close button
3. Sign off the account
The problem is that the dialog was closed, but it was not removed from
the list of open dialogs. So purple_notify_close_with_handle() tried
to close it again.

diff -urN pidgin-2.5.2.orig/pidgin/gtknotify.c pidgin-2.5.2/pidgin/gtknotify.c
--- pidgin-2.5.2.orig/pidgin/gtknotify.c	2008-08-29 21:53:56.000000000 -0400
+++ pidgin-2.5.2/pidgin/gtknotify.c	2008-11-21 14:55:56.000000000 -0500
@@ -703,7 +703,7 @@
 	gtk_widget_grab_focus(button);
 
 	g_signal_connect_swapped(G_OBJECT(button), "clicked",
-							 G_CALLBACK(gtk_widget_destroy), window);
+							 G_CALLBACK(formatted_close_cb), window);
 	g_signal_connect(G_OBJECT(window), "key_press_event",
 					 G_CALLBACK(formatted_input_cb), NULL);
 
